test/unit/common/test_range2d.cpp

```diff
@@ -113,3 +113,81 @@ TEST(DoubleArgEmptyRange2D, Index2D) {
 TEST(DoubleArgEmptyRange2D, Size2D) {
   ::test_double_arg_empty(Size(0, 0));
 }
+
+/// This is a very trivial test fixture that creates the following range2D
+///
+/// +---------+--------+---------+
+/// | (1, 1)B | (1, 2) | (1, 3)E |
+/// | (2, 1)  | (2, 2) |         |
+/// | (3, 1)  | (3, 2) |         |
+/// +---------+--------+---------+
+///
+/// {(1, 1) == range.begin(), (2, 1), (3, 1), (1, 2), (2, 2), (3, 2), (1, 3) == range.end()}
+struct IteratorTest : public ::testing::Test {
+  using range2D_t = decltype(iterate_range2d(::Index{0, 0}, ::Size{0, 0}));
+
+  const range2D_t range = iterate_range2d(::Index{1, 1}, ::Size{3, 2});
```

Comment:
I'm a bit confused about this part, why is (1,3) part of the range?

Reply:
Let's check together if the assumptions are right, maybe it's just the documentation that it is misleading.

https://github.com/eth-cscs/DLA-Future/blob/1232535cd317667dcda8ecb75495ffff2d641007/test/unit/common/test_range2d.cpp#L129

This defines a 3x2 range2D starting from (1, 1), so it is a range that contains:
```
/// +--------+--------+
/// | (1, 1) | (1, 2) |
/// | (2, 1) | (2, 2) |
/// | (3, 1) | (3, 2) |
/// +--------+--------+
/// 
/// {(1, 1), (2, 1), (3, 1), (1, 2), (2, 2), (3, 2)}
```

But, since we are dealing with iterators, we know that they work with the logic [begin, end), so the end is the first one past the end of the actual range. From an implementation detail (and this may be considered an error), we know that the "end" will be (1, 3).

So, (1, 3) is not part of the range, I was just depicting it in the documentation with the aim of (hopefully) give a better overview of the various arithmetic operations in the next tests.

Hope this helps you and I look forward to your comments/suggestions on how to improve it.

Reply:
I see, it's OK to leave it as it is.